sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -447,8 +475,10 @@ object AggUtils {
       val aggregateExpressions = functionsWithoutDistinct.map(_.copy(mode = PartialMerge))
       val aggregateAttributes = aggregateExpressions.map(_.resultAttribute)
       MergingSessionsExec(
-        requiredChildDistributionExpressions = None,
-        requiredChildDistributionOption = Some(restored.requiredChildDistribution),
+        requiredChildDistributionExpressions = Some(groupingWithoutSessionAttributes),
+        isStreaming = true,
+        // This will be replaced with actual value in state rule.
```

Comment:
let's link the state rule class name here?

Reply:
We don't create a dedicate class for state rule. See https://github.com/apache/spark/blob/a30575e91e82e9f4394add40f1ba9265eb1f7819/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala#L105-L209